### PR TITLE
PSR-1278 Added land or property dispoal view only page

### PR DESCRIPTION
**Pull request by westwater**

### CODE REVIEW

File: app/models/ViewOnlyViewModel.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import java.time.LocalDateTime
+
+case class ViewOnlyViewModel(
```
Comment (westwater) [inline app/models/ViewOnlyViewModel.scala:21]: avoids passing many optionals into a viewmodel when in ViewOnlyMode

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -20,4 +20,17 @@
 
 .autocomplete__wrapper {
   width: 66.66% !important;
+}
+
+a[href="none"] {
```
Comment (westwater) [inline app/assets/stylesheets/application.scss:25]: required for the "no" view only pages - devs will be having a chat tomorrow to discuss whether we take this solution or an alternative forward

**Review by narendrapaduchuri — APPROVED** [review]

File: app/controllers/nonsipp/landorpropertydisposal/LandOrPropertyDisposalListController.scala [hunk 3]
```diff
@@ -154,19 +246,59 @@ object LandOrPropertyDisposalListController {
...
+    srn: Srn,
+    mode: Mode,
+    addressesWithIndexes: List[((Max5000, List[Max50]), Address)],
+    viewOnlyViewModel: Option[ViewOnlyViewModel],
+    schemeName: String
+  ): List[ListRow] =
+    addressesWithIndexes match {
+      case Nil =>
+        List(
+          ListRow.viewNoLink(
```
Comment (westwater) [inline app/controllers/nonsipp/landorpropertydisposal/LandOrPropertyDisposalListController.scala:259]: This is the "no" page implementation

File: app/views/ListView.scala.html [hunk 4]
```diff
@@ -66,8 +66,11 @@ <h1 class="govuk-heading-l">@renderMessage(viewmodel.optViewOnlyDetails.get.head
                             .withVisuallyHiddenText(change.hiddenText.toMessage)).toList ++
                                     value.remove.map(remove => ListWithActionsActionViewModel(messages("site.remove"), remove.url)
                                     .withVisuallyHiddenText(remove.hiddenText.toMessage)).toList ++
-                                    value.view.map(view => ListWithActionsActionViewModel(messages("site.view"), view.url)
-                                    .withVisuallyHiddenText(view.hiddenText.toMessage)).toList
+                                    value.view.map {
+                                        case ListRowLink(url, hiddenText) => ListWithActionsActionViewModel(messages("site.view"), url)
+                                        .withVisuallyHiddenText(hiddenText.toMessage)
+                                        case ListRowNoLink(text) => ListWithActionsActionViewModel.noLink(renderMessage(text))
```
Comment (westwater) [inline app/views/ListView.scala.html:72]: change to allow ListView to display rows without links (like for the "no" page)

**Pull request merged**